From 97e9f1a11717506574e23c64ea0d82b6f8477f68 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Mon, 16 May 2016 18:40:30 +0800
Subject: [PATCH 0490/1722] drm/amd/dal: defer clean-up link in stream status

It is needed to make sure link is available in stream
status until disabled stream is finished. On some
platforms link is used in MST helpers

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index 8f3fedb..5934447 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -1669,14 +1669,13 @@ void core_link_disable_stream(struct pipe_ctx *pipe_ctx)
 {
 	struct core_dc *core_dc = DC_TO_CORE(pipe_ctx->stream->ctx->dc);
 
-	pipe_ctx->stream->status.link = NULL;
 	if (pipe_ctx->signal == SIGNAL_TYPE_DISPLAY_PORT_MST)
 		deallocate_mst_payload(pipe_ctx);
 
 	core_dc->hwss.disable_stream(pipe_ctx);
 
-	pipe_ctx->stream->status.link = NULL;
-
 	disable_link(pipe_ctx->stream->sink->link, pipe_ctx->signal);
+
+	pipe_ctx->stream->status.link = NULL;
 }
 
-- 
2.7.4

